### PR TITLE
Remove lots of commented out code

### PR DESCRIPTION
**Pull request by kotarou3**

### CODE REVIEW

**Comment by DanUgore** [comment]
Eh. I don't see why the debug ones cant stay in. Removing the '-message' ones makes sense, since those are supposed to be client-side anyway.

**Comment by kotarou3** [comment]
The alternative with the debug ones would be to uncomment them, since they aren't shown by default anymore either.

**Comment by DanUgore** [comment]
They don't need to be uncommented in the deployed code. But it makes transitioning from that into debugging smoother. The very marginal "benefit" of having a few less lines in the codebase doesn't outweigh the ease of just uncommenting the line when you want to debug something.

**Comment by Slayer95** [comment]
Agreed that the ``this.debug()`` lines should remain...

Note that ``this.debug()`` <i>does</i> stuff even outside of custom game, e.g. in
https://github.com/Zarel/Pokemon-Showdown/pull/2473/files#diff-8eafa7b779907ba691f7e2da9615ab5dL2496

(I'd be very surprised if V8 obviates the stringification).

I take the chance to ask what do you think of writing stuff like:
``this.debug('status: %json', thing)``

**Comment by kotarou3** [comment]
`this.debug()` stuff added back and left commented out

File: battle-engine.js [hunk 1]
```diff
@@ -2264,10 +2254,6 @@ Battle = (() => {
 	 *   they're useful for functions called by the event handler.
 	 */
 	Battle.prototype.runEvent = function (eventid, target, source, effect, relayVar, onEffect, fastExit) {
-		// if (Battle.eventCounter) {
-		// 	if (!Battle.eventCounter[eventid]) Battle.eventCounter[eventid] = 0;
-		// 	Battle.eventCounter[eventid]++;
-		// }
```
Comment (Zarel) [inline battle-engine.js]: This is legitimately useful debug code :|

**Comment by Zarel** [comment]
Honestly, other than the crashlogger thing, all code you're removing is debug code...

File: data/scripts.js [hunk 2]
```diff
@@ -2812,7 +2812,6 @@ exports.BattleScripts = {
 		} while (moves.length < 4 && movePool.length);
 
 		// any moveset modification goes here
-		//moves[0] = 'safeguard';
```
Comment (ascriptmaster) [inline data/scripts.js]: This is actually an example so other people know how to do moveset modification

**Comment by kotarou3** [comment]
A lot of stuff is really old debug code. The problem really is that often it ends up unintentionally diverging from the surrounding code. For example: https://github.com/Zarel/Pokemon-Showdown/pull/2473/files#diff-8d651dbad6d09eee255c9814099a1228L932

Maybe some of the `console.log`s could be put behind a `Config.quietconsole` config?

**Comment by Zarel** [comment]
Our refactor commits tend to refactor the commented debug code, too. Sometimes it doesn't but that's rare.

**Comment by Zarel** [comment]
Closing, unmaintained, inconclusive...

**Pull request closed without merging**